common/persistence/sql/sqlplugin/mysql/events.go

```diff
@@ -46,6 +46,8 @@ const (
 	getHistoryTreeQuery = `SELECT branch_id, data, data_encoding FROM history_tree WHERE shard_id = ? AND tree_id = ? `
 
 	deleteHistoryTreeQuery = `DELETE FROM history_tree WHERE shard_id = ? AND tree_id = ? AND branch_id = ? `
+
+	getAllHistoryTreeQuery = `SELECT branch_id, data, data_encoding FROM history_tree WHERE (shard_id = ? AND tree_id = ? AND branch_id > ?) OR (shard_id = ? AND tree_id > ?) OR (shard_id > ?) ORDER BY shard_id, tree_id, branch_id LIMIT ?`
```

Comment:
I believe you have noticed that listTaskList query doesn't use `shard_id>?`, but instead it uses a forloop to iterate over all shards. The reasons is that we were trying to implement in a way that can work for some sharded/partitioned MySQL. In partitioned MySQL, all shards are independent hence we cannot query across shards directly.

However, there is no real usecase of partitioned MySQL...so I won't force you to implement it. And it won't work for your getOrphanTask query which will have to do corss-partition query. 

Would you mind add a TODO here linking to issue that I just opened? https://github.com/uber/cadence/issues/4064

Reply:
The comment may be better put in sqlTaskManager